app/services/ui_notifications/rss_notifications_checker.rb

```diff
@@ -0,0 +1,50 @@
+require 'rss'
+require 'date'
+
+module UINotifications
+  class RssNotificationsChecker
+    def initialize(options = [])
+      @url = options[:url] || 'https://theforeman.org/feed.xml'
+      @latest_posts = options[:latest_posts] || 3
+      @force_repost = options[:force_repost] || false
+      @audience = options[:audience] || Notification::AUDIENCE_GLOBAL
+    end
+
+    def deliver!
+      feed = RSS::Parser.parse(@url, false)
```

Comment:
if it fails to reach out (lets say a proxy, network transient error etc) is it being logged some where? should we?

Reply:
Good catch - I have included a rescue in the job itself for this situation, it will be logged to the background logger